src/components/views/ViewReusablePaymentRequest.tsx

```diff
@@ -1,44 +1,79 @@
 import { usePaymentRequests } from '@/hooks/cashu/usePaymentRequests';
 import QRCode from 'qrcode.react';
-import { useEffect, useState } from 'react';
+import { useCallback, useEffect, useState } from 'react';
 import ClipboardButton from '../buttons/utility/ClipboardButton';
 import { useCashuContext } from '@/hooks/contexts/cashuContext';
 import { decodePaymentRequest } from '@cashu/cashu-ts';
 import useMintlessMode from '@/hooks/boardwalk/useMintlessMode';
 import { Spinner } from 'flowbite-react';
+import { usePolling } from '@/hooks/util/usePolling';
+import { useCashu } from '@/hooks/cashu/useCashu';
 
-const ViewReusablePaymentRequest = () => {
-   const [paymentRequest, setPaymentRequest] = useState('');
-   const { fetchPaymentRequest, fetchingPaymentRequest } = usePaymentRequests();
+interface ViewReusablePaymentRequestProps {
+   onSuccess?: () => void;
+}
+
+const ViewReusablePaymentRequest = ({ onSuccess }: ViewReusablePaymentRequestProps) => {
+   const { fetchPaymentRequest, lookupLastPaid } = usePaymentRequests();
+   const [paymentRequest, setPaymentRequest] = useState<{ pr: string; id: string } | null>(null);
+   const [lastPaid, setLastPaid] = useState<string | null>(null);
    const { activeWallet, activeUnit } = useCashuContext();
+   const { handleClaimToSourceMint } = useCashu();
    const { isMintless } = useMintlessMode();
 
+   const pollPaymentRequest = useCallback(async () => {
+      if (!paymentRequest?.id) return;
+
+      try {
+         const res = await lookupLastPaid(paymentRequest.id);
+
+         if (!lastPaid) {
+            /* First time checking - initialize lastPaid */
+            setLastPaid(res.lastPaid || new Date(0).toISOString());
+            return;
+         }
+
+         if (res.lastPaid && res.lastPaid > lastPaid) {
+            /* payment request was paid since this component has been open */
+            handleClaimToSourceMint(res.token);
```

Comment:
I am not sure I understand what are we doing here?

Reply:
`lastPaid` on the response will be null if the cashu request was never paid.
 If the request was paid and it was paid after we opened this ViewReusablePaymentRequest view, then we know that a new token was received. We got a new token so we claim it.
 
handleClaimToSource mint performs the necessary logic to claim the token to the mint that it is from. If the token is from stablenut, then we will swap it with a fresh token from stablenut, and store the new proofs.